### PR TITLE
Move remaning compile-fail tests that are rejected by the parser to pars...

### PR DESCRIPTION
**Pull request by fhahn**
After making `rustc` fail on errors at a stop point, like `-Z parse-only`, in #22117, the files in this PR also fail during the parse stage and should be moved as well. Sorry for spliting this move up in two PRs.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Hm I don't actually think we're running any parse-fail tests at all! Looking at the most recent output for a successful test run there are no mentions of any `parse-fail` tests. I think that [this change](https://github.com/rust-lang/rust/pull/22011/files#diff-ebe2626c4d404bbab7b4a1790121fff7R472) was only meant for re-running `compiletest`, the actual crawling of a directory comes from the [`--src-base`](https://github.com/rust-lang/rust/blob/01db9a46af72383811f39091be70e9721aaa159a/mk/tests.mk#L651) argument and the logic may need to be tweaked to have "two source bases" for this.

Can you update this PR as well to be sure to run all the just-moved `parse-fail` tests and make sure that `make check` runs all tests?

**Comment by Manishearth** [comment]
You might want to add https://github.com/Manishearth/rust/commit/19d8d3e53f7449c1b82765d4968477bc6e9f8687 to this PR too.

**Comment by fhahn** [comment]
@alexcrichton damn, sorry! Should I modify `compiletest.rs` to accept more than a single path as `--src-base` or should I add a separate make target for parse-fail?

**Comment by alexcrichton** [comment]
Whatever is easiest is fine by me, I'd probably go with multiple `--src-base` arguments as I think getopts already handles it. Also no worries, don't sweat it!

**Comment by fhahn** [comment]
I've opted to add another `pfail` target, because I was not sure how to add multiple `--src-base` arguments in the Makefile in a generic way.

However I am not sure which part of the Makefile is responsible for copying the test files to the test directory. Any idea what I could have missed or pointers where I could start looking?

**Comment by alexcrichton** [comment]
When you say copying test files, what are you referring to? I think your changes here should be sufficient for running the tests.

**Comment by fhahn** [comment]
I got it wrong, I just had to create the ´test/parse-fail´ directory in the configure script. It should run the parse-fail tests now. But `compiletest` still prints `[compile-fail] parse-fail/....`, because I am reusing this code. Could this be somehow confusing?

**Comment by fhahn** [comment]
@Manishearth the test case you reference fails during macro expansion (or later, I am not sure), because the `include!` has to be expanded first. So I think adding it to `parse-fail` would not be such a good idea, because I would like to add the `-Z parse-only` option to all `parse-fail` tests  and the LALR parser does not handle macro expansion (although we could include some extra code to also parse files included with `include!`)

**Comment by brson** [comment]
@fhahn I would prefer if you could add a ParseFail mode to compiletest if it's easy to do.

**Comment by fhahn** [comment]
@brson I've updated the code.

File: mk/tests.mk [hunk 1]
```diff
@@ -173,12 +173,12 @@ check-notidy: cleantmptestlogs cleantestlibs all check-stage2
...
 	$(foreach crate,$(TEST_TARGET_CRATES),check-stage2-$(crate)) \
 	check-stage2-rpass check-stage2-rpass-valgrind \
-	check-stage2-rfail check-stage2-cfail check-stage2-rmake
+	check-stage2-rfail check-stage2-cfail check-stage2-pfail check-stage2-rmake
 	$(Q)$(CFG_PYTHON) $(S)src/etc/check-summary.py tmp/*.log
 
 # Only check the 'reference' tests: rpass/cfail/rfail/rmake.
 check-ref: cleantestlibs cleantmptestlogs check-stage2-rpass check-stage2-rpass-valgrind \
-	check-stage2-rfail check-stage2-cfail check-stage2-rmake
+	check-stage2-rfail check-stage2-cfail check-stage2-pfail ccheck-stage2-rmake
```
Comment (alexcrichton) [inline mk/tests.mk]: ccheck typo

**Comment by alexcrichton** [comment]
Thanks @fhahn, looks good to me! Just a small typo and otherwise r=me.

Also, just to confirm, but you see the output of the tests in `make check`?

**Comment by fhahn** [comment]
@alexcrichton I have fixed the type and now parse-fail and compile-fail tests show up when running `make check`

File: mk/tests.mk (hunk 1 above)
Comment (alexcrichton) [inline mk/tests.mk]: (ping typo)

**Comment by alexcrichton** [comment]
(I think the typo is still there)

**Comment by fhahn** [comment]
Argh yes, I somehow forgot to push the updated commit. Sorry for that!

**Comment by alexcrichton** [comment]
@bors: r+ 6824f1365d2d1dc967c92bec11c44920bbf3da68

(I'll fix the typo in the upcoming rollup

**Pull request merged**